### PR TITLE
kube ci testing

### PR DESCRIPTION
**Pull request by jrhizor**
Runs acceptance tests in kube on CI. 

@cgardens I know we talked about if this is something we should do here: I'm going to time it. If it isn't super long I think we should have it for now. Otherwise Kube-incompatible changes will be really hard to detect. This was actually a problem when testing the Kube stuff and merging in `master` changes.

### CODE REVIEW

**Comment by jrhizor** [comment]
For some reason it's having trouble running on Github Actions. It works locally for me. I'll debug today.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.testcontainers.shaded.org.apache.commons.lang.RandomStringUtils;
+
+/**
+ * These tests are a limited form of the tests in {@link AcceptanceTests}. They are only testing for
+ * a lack of API failures. For now, we aren't re-using {@link AcceptanceTests} because we would need
```
Comment (sherifnada) [inline airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java:70]: Is the issue that AcceptanceTests uses a docker container? If so, would it make sense to use ERIO there?

**Review by sherifnada — APPROVED** [review]
awesome! 

I think it'd save us work in the future to use the same AcceptanceTest if possible. I don't fully understand why that's not possible from the docstring.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java (hunk 1 above)
Comment (jrhizor) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java:70]: It uses a Docker container that exists outside of Kube, so we would have to either provide a tunnel for the Kube service to access the localhost (non-Kube) DB or we would need to pakcage the docker operations into Kube. Longer term we should do both. Short term this seemed sufficient.

Comment (sherifnada) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java:70]: Should we just stop using a container db and use ERIO instead?

Comment (cgardens) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java:70]: I think I would prefer Sherif's suggestion here, if the only change is that we are switching out the source, then I would prefer we use the same AcceptanceTest class for both kube and docker versions.  I'm fine with us just using exchange rate api for both of them. AcceptanceTest is a big class to just copy and paste. We also lose the value of having acceptance tests a little bit if kube and docker have different acceptance tests (or if those acceptance tests drift apart meaningfully, which i assume will happen.) 

If I'm missing something more complex in here, lmk.

**Review by cgardens — CHANGES_REQUESTED** [review]
looks good. 

requesting changes because it seems to me that we should just change AcceptanceTest to just use exchange rate api and have both the docker and kube version use the same test suite. more detailed comment about this below in that that thread. once that is sorted, it's an approve.

> @cgardens I know we talked about if this is something we should do here: I'm going to time it. If it isn't super long I think we should have it for now. Otherwise Kube-incompatible changes will be really hard to detect. This was actually a problem when testing the Kube stuff and merging in master changes.

cool. works for me.

File: airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java (hunk 1 above)
Comment (jrhizor) [reply airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTestsKube.java:70]: I think it's much more valuable to have Postgres as a source for incremental testing, more data types, incremental vs full refresh, the potential for customization of data type transformations, etc. 

The new class is 1/2 the size of the other. 

Really I think of this is a stopgap until we add support for different types of container interactions (docker and kube) in the main acceptance test class. Then we could run it in both.

**Review by cgardens — APPROVED** [review]
@jrhizor and I talked offline. He's going to look into one other approach to try to get the tests to work with the docker container. Seems like there are some options ([1](https://stackoverflow.com/a/55762427/4195169), [2](https://minikube.sigs.k8s.io/docs/handbook/host-access/#hostminikubeinternal)) with minikube that might "just work". If they don't we'll punt on this and stick with what he's got.

**Comment by jrhizor** [comment]
Alright I'm at the end of this. It's a bit of a compromise.

It's using the same test class, but it has optional checks for target file creation. 

It can access the non-kube docker db, but it can't access the mounts without a bit more work.

**Pull request merged**